site/content/integrate/apps/interactivity/dynamic-form-example/_index.md

```diff
@@ -0,0 +1,544 @@
+weight: 5
+---
+
+### Overview
+
+Each of these examples will be using a modal form to help the user create a ticket in an external system such as Jira or Zendesk. The form will get a little more complex in each example. In this guide we'll cover:
+
+- How to render an item in the post dot menu
+- How to open a modal when the post menu item is clicked
+- How to create a form with:
+    - Text fields
+    - Static select fields
+    - Dynamic autocomplete select fields
+- How to redefine the form based on user interaction
+- Using `context` and `expand`:
+    - How to access the associated post after interacting with the post menu
+    - How to access other entities in the user's context such as the current `Team` or `Channel`
+
+### Setup
+
+These examples assume you have a Node.js App set up using [express](https://github.com/expressjs/express), like the [quick start example]({{< ref "quick-start-js" >}}), or the [example](https://github.com/mattermost/mattermost-plugin-apps/blob/master/dev/node_app/src/app.ts) from the App framework [development environment](https://github.com/mattermost/mattermost-plugin-apps/tree/master/dev).
+
+The examples on this page use types from the [mattermost-redux](https://github.com/mattermost/mattermost-redux) library, namely the App-specific types in [types/apps.ts](https://github.com/mattermost/mattermost-redux/blob/master/src/types/apps.ts); Mattermost types `Post`, `UserProfile`, and `Team`; and the Mattermost JavaScript client. See the end of the guide for the import paths for each of these.
+
+### Manifest
+
+When our App is installed, we will return this Manifest to define our App:
+
+```ts
+app.get('/manifest', (req, res) => {
+    const manifest = {
+        app_id: 'ticket-form-example',
+        display_name: "Create tickets using forms",
+        homepage_url: 'https://github.com/mattermost/mattermost-plugin-apps',
+        app_type: 'http',
+        icon: 'icon.png',
+        root_url: `http://localhost:3000`,
+        requested_permissions: [
+            'act_as_bot', // This is necessary to create public posts as our bot
+        ],
+        requested_locations: [
+            '/post_menu', // This is necessary to render elements in the post menu
+        ],
+    } as AppManifest;
+
+    res.json(manifest);
+});
+```
+
+With our server running, we can then install our App in Mattermost using this command:
+
+`/apps install http http://localhost:3000/manifest`
+
+### Bindings
+
+The App will receive a request to provide UI elements to show to the user, whenever the user visits a different channel. This allows us to dynamically show different UI elements or slash commands in each channel. We need to handle this request being sent to the `/bindings` endpoint.
+
+```ts
+app.post('/bindings', (req, res) => {
+    const call: AppCallRequest = req.body;
+
+    // Use the user id and channel id to dynamically return certain UI elements. Though for this example, we won't use these value.
```

Comment:
1/5 - might make sense to not even mention them (if we don't use them - can just confuse the developer) or augment the documentation with an example that actually uses them.